src/amp.js

```diff
@@ -121,6 +122,7 @@ if (shouldMainBootstrapRun) {
 
         maybeValidate(self);
         makeBodyVisible(self.document);
+        detectAutoLightbox(ampdoc);
```

Comment:
This should happen in your extension's initialization process, not here.

Reply:
There needs to be a path to the extension from the runtime, this function only checks for existence of `amp-img` and then includes `amp-auto-lightbox`. This is essentially a runtime feature.

Reply:
I agree, this shouldn't be happening here. If the publisher wants auto-lightbox treatment, they should be including `amp-auto-lightbox.js`.

Reply:
This is supposed to be backported to documents that do *not* include the extension. /cc @aghassemi 

For internal details see go/amp-auto-lightbox. Public I2I TBD.

Reply:
@choumx @jridgewell: the whole feature is lightboxing images without publisher's involvements (because they don't do it and end user suffers as the result). We will be running experiments and monitoring perf metrics as this progresses. Initially we will be starting with a limited environment to enable this feature (likely just AMP pages in SERP viewer that are of type Article). This PR is purely to put it behind an experiment flag so we can enable manually and start looking at pages to refine the heuristics.

Reply:
(I also assume tree shaking is smart enough not to include all of auto-lightbox.js here in v0. `detectAutoLightbox` simply checks for couple of conditions and then dynamically loads another extension that would take care of the rest of heuristics)

Reply:
Have we considered fetching the extension binary at runtime for experiment-enabled users? Or injecting the extension script via cache transform?

Reply:
> Have we considered fetching the extension binary at runtime for experiment-enabled users?

That's exactly what `detectLightbox` does unless I'm missing something.

> Or injecting the extension script via cache transform?

That would be a nice improvement over this, I don't see how they're mutually exclusive?